pytest_splunk_addon/cim_tests/test_generator.py

```diff
@@ -270,7 +270,14 @@ def generate_recommended_fields_tests(self):
                     model = model.replace(" ", "_")
                     if datasets:
                         datasets = [dataset.replace(" ", "_") for dataset in datasets]
-
+                    common_fields = list(
+                        set(event.requirement_test_data["cim_fields"].keys())
+                        & set(event.requirement_test_data["missing_recommended_fields"])
+                    )
+                    if common_fields:
+                        LOGGER.warning(
+                            f"Common fields found in both cim_fields and missing_recommended_fields for {event.sample_name} sample = {common_fields}"
+                        )
```

Comment:
Shouldn't we log an error and stop the execution if we something like this?

This would be technically a breaking change but I think we are fine with it - this would need to be documented properly in the PSA documentation.

Reply:
So if it is preset in both, then the tests would work as expected as it is [appending](https://github.com/splunk/pytest-splunk-addon/blob/develop/pytest_splunk_addon/cim_tests/test_generator.py#L274
) both missing_recommended_fields and cim_fields at the end, so nothing would change from execution perspective.

Reply:
So actually functionally it doesn't change anything, beside suggestion that there are fields defined as both missing_recommended_fields and cim_fields right?
If so, then if we aim for "clean" samples, then I'd break tests execution, as I think nobody will really pay an attention to warning somewhere in the middle of PSA tests execution.

Reply:
> So actually functionally it doesn't change anything, beside suggestion that there are fields defined as both missing_recommended_fields and cim_fields right?

Yes

> If so, then if we aim for "clean" samples, then I'd break tests execution, as I think nobody will really pay an attention to warning somewhere in the middle of PSA tests execution.

Breaking test execution for something like this feels a bit unnecessary, but I do agree with part that warning would be hard to find between so many logs of PSA.